docs/Using-Fleet/Mobile-device-management.md

```diff
@@ -2,14 +2,58 @@
 
 MDM features are not ready for production and are currently in development. These features are disabled by default.
 
-MDM features allow you to mange macOS updates and macOS settings on your hosts.
+MDM features allow you to manage macOS updates and macOS settings on your hosts.
 
-To use MDM features you have to connect Fleet to Apple Push Certificates Portal:
+## Apple Push Notification Service (APNs)
 
-1. In the Fleet UI, head to the **Settings > Integrations > Mobile device management (MDM)** page. Users with the admin role can access the settings pages.
+To use MDM features you have to connect Fleet to Apple Push Certificates Portal. This can be done via the Fleet UI or the `fleetctl` command-line interface. The result is the same regardless of the method used, you end up with 4 files: an APNs certificate and private key files along with the Simple Certificate Enrollment Protocol (SCEP) certificate and private key files. Make sure you store them securely as you will need them to configure your Fleet instances afterwards (and take note of the email and organization used to generate them, as you will need to use the same ones during renewal). For _renewal_ of an existing but expired (or soon to expire) APNs certificate, see the [APNs Renewal section](#apns-renewal) below.
 
+Via the Fleet UI:
+1. Head to the **Settings > Integrations > Mobile device management (MDM)** page. Users with the admin role can access the settings pages.
 2. Follow the instructions under **Apple Push Certificates Portal**.
 
+Via the `fleetctl` CLI:
+1. Run `fleetctl generate mdm-apple --email <email> --org <org>`.
+2. Follow the on-screen instructions.
+
+### APNs Renewal
+
+The APNs certificate is typically valid for a year. You can see the certificate's renewal date and other important APNs information via the Fleet UI or the `fleetctl` command-line interface:
+
+Via the Fleet UI:
+1. Head to the **Settings > Integrations > Mobile device management (MDM)** page. Users with the admin role can access the settings pages.
+2. Look at the **Apple Push Certificates Portal** section.
+
+Via the `fleetctl` CLI:
+1. Run `fleetctl get mdm-apple`.
+2. Look at the on-screen information.
+
+If the certificate is expired or about to expire, you must renew it by using the same command that can be used to generate it the first time, `fleetctl generate mdm-apple --email <email> --org <org>`. The Fleet UI cannot be used for renewal at the moment. Note that you must make sure that you use the same email and organization as when the certificate was generated. One **important difference** when renewing a certificate is that in the [Apple portal to get the new certificate](https://identity.apple.com), you must click on the _Renew_ button so that the same APNs topic is reused.
```

Comment:
```suggestion
How to renew the certificate if it's expired or about to expire: 

1. Run the `fleetctl generate mdm-apple --email <email> --org <org>` command. 

2. Sign in to [Apple Push Certificates Portal](https://identity.apple.com) using the same Apple ID you used to get your original certificate. If you don't use the same Apple ID, you will have to turn MDM off and back on for all macOS hosts.

3. In the **Settings > Integrations > Mobile device management (MDM)** page, under Apple Push Certificates portal, find the serial number of your current certificate. In Apple Push Certificates Portal, click  **Renew** next to the certificate that has the matching serial number. If you don't renew and get a new certificate, you will have to turn MDM off and back on for all macOS hosts.
```

I broke this out into several steps and added information about finding the serial number in the Fleet UI.

I have the same question about `--org` and `--email`. Does using the same matter?

Reply:
I'm sure for --email, not for --org, your text looks good (and clearer) to me.

Reply:
> I'm sure for --email, not for --org

@mna hmm, I'm confused. 

Let's say I use `noah@example.com` to generate my original CSR. I log in to Apple Push Certs Portal with this email. Later, when I renew, I set `--email` to `bob@example.com` I still login to Apple with `noah@example.com`.

Will renewing work? (I don't have to re-enroll hosts).

Or, during renewal, do I have to use `noah@example.com` to generate the CSR? To prevent having to re-enroll hosts.

Reply:
The below is from the following Slack thread (internal): https://fleetdm.slack.com/archives/C03C41L5YEL/p1674586568847999?thread_ts=1674586435.140469&cid=C03C41L5YEL

> Let's say I use noah@example.com to generate my original CSR. I log in to Apple Push Certs Portal with this email. Later, when I renew, I set --email to bob@example.com I still login to Apple with noah@example.com.

Martin

You will probably see the certificate to renew, but my understanding is that after renewal, you may not find the new cert there (it will be in the bob account, or it may even fail to renew, I don't know). According to https://docs.jamf.com/jamf-now/documentation/Determining_the_Apple_ID_Used_for_APNs_Setup.html , the certificate can be found under the account of the certificate owner AFAICT.


All in all, I think it's better to recommend using the same? Though surely there's a process to follow if the old ID is not at the organization anymore.


Noah

> after renewal, you may not find the new cert there (it will be in the bob account, or it may even fail to renew, I don’t know).

Ah, got it. Yeah I’m not sure either. Can we test this?


Noah

For now, adding back the recommendation to use the same email makes sense to me. This way we can get the docs merged

Martin

> Can we test this?

When we QA this we should make sure to cover that. I left a comment to @xpkoala on the ticket, I'll add that part to it (renew with a different email).